### PR TITLE
test: add changes to improve test coverage for httpx

### LINKED ISSUE #2159: Improve httpx instrumentation test coverage
The coverage of the httpx instrumentation doesn't meet the expected 85% as calculated by simplecov.

We should add additional test cases so that this gem is being adequately test after every change. Once the coverage meets the required level the coverage override defined in the github Workflow needs to be removed.

### PR DESCRIPTION
**Pull request by Oliver-ke**
## Description

Fixes #2159

Adds unit tests for `HttpHelper` and `Instrumentation`, and edge-case tests for url.query and server.port
attributes in stable/dup plugin tests. 

Merged coverage across all appraisals goes from `~59%` to `97.8%`.

### CODE REVIEW

**Review by xuan-cao-swi — APPROVED** [review]

File: instrumentation/httpx/test/instrumentation/httpx/instrumentation_test.rb [hunk 1]
```diff
@@ -25,6 +23,54 @@
...
+      _(instrumentation.determine_semconv).must_equal 'dup'
+    end
+  end
+
+  describe '#emit_old_semconv_deprecation_warning' do
+    it 'emits a deprecation warning' do
+      OpenTelemetry.stub(:logger, Logger.new(StringIO.new)) do
+        expect(OpenTelemetry.logger).to receive(:warn).with(/deprecated/)
+        instrumentation.emit_old_semconv_deprecation_warning('old')
+      end
```
Comment (thompson-tomo) [inline instrumentation/httpx/test/instrumentation/httpx/instrumentation_test.rb]: Can you look at this, I think it should be 

```suggestion
      allow(OpenTelemetry).to recieve(:logger).and_return(Logger.new(StringIO.new))
      expect(OpenTelemetry.logger).to receive(:warn).with(/deprecated/)
      instrumentation.emit_old_semconv_deprecation_warning('old')
```

**Comment by arielvalentin** [comment]
@HoneyryderChuck please review this PR.

Also, please sign up join the open-telemetry community so that you can receive notifications for reviews through codeowners

File: instrumentation/httpx/test/instrumentation/httpx/instrumentation_test.rb (hunk 1 above)
Comment (Oliver-ke) [reply instrumentation/httpx/test/instrumentation/httpx/instrumentation_test.rb]: Good catch on the consistency,  mixing Minitest stub with rspec-mocks is noisy.
I noticed a typo, `recieve` instead of `receive`, in the suggestion. 

I'll push:

```
allow(OpenTelemetry).to receive(:logger).and_return(Logger.new(StringIO.new))
expect(OpenTelemetry.logger).to receive(:warn).with(/deprecated/)                                                                                                                  
instrumentation.emit_old_semconv_deprecation_warning('old') 
```

**Comment by Oliver-ke** [comment]
> @HoneyryderChuck please review this PR.
> 
> Also, please sign up join the open-telemetry community so that you can receive notifications for reviews through codeowners

Thanks, I've joined Slack #otel-ruby channel

**Review by HoneyryderChuck — APPROVED** [review]

File: instrumentation/httpx/test/instrumentation/httpx/instrumentation_test.rb [hunk 2]
```diff
@@ -25,6 +23,53 @@
...
+
+  describe 'compatible' do
+    it 'returns true when HTTPX version is >= 1.6.0' do
+      _(instrumentation.compatible?).must_equal true
+    end
+  end
+
+  describe 'present' do
+    it 'returns truthy when HTTPX is defined' do
+      assert instrumentation.present?
```
Comment (thompson-tomo) [inline instrumentation/httpx/test/instrumentation/httpx/instrumentation_test.rb]:
```suggestion
      assert_predicate instrumentation, :present?
```

**Comment by arielvalentin** [comment]
@Oliver-ke it seems that after merging this PR, code coverage is failing on main:
https://github.com/open-telemetry/opentelemetry-ruby-contrib/actions/runs/24725752926/job/72326370847

May I ask you to take a look at this?

**Comment by Oliver-ke** [comment]
> @Oliver-ke it seems that after merging this PR, code coverage is failing on main: https://github.com/open-telemetry/opentelemetry-ruby-contrib/actions/runs/24725752926/job/72326370847
> 
> May I ask you to take a look at this?

@arielvalentin Strange, looks like it's due to a reduction in expected coverage. However, I see a higher coverage locally. 
I'm happy to look into this more on a new PR
